### PR TITLE
Run E2E AWS single node variant on machine-config-operator presubmit

### PR DESCRIPTION
**Pull request by omertuc**
Based on #15187

### CODE REVIEW

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:436]: can you add optional: false

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:425]: if this is expected to fail  let's set always_run:false

Comment (kikisdeliveryservice) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:425]: this will make it manual trigger

**Review by kikisdeliveryservice — CHANGES_REQUESTED** [review]
You also need to update:

https://github.com/openshift/release/blob/master/ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-master.yaml

see: https://github.com/openshift/release/commit/7b492163aa6d364a7eb8812daaaefc1abe176f95 for example

**Comment by kikisdeliveryservice** [comment]
We don't manually generate those test files anymore... Basically you update the master.yaml file I linked to above then run:

$ make update

then fix the fields as i suggested..

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml
Comment (omertuc) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:436]: Actually wanted to add `optional: true` and `always_run: false` but forgot. Will fix that.

This is because for now those tests don't pass because of some OpenShift components not being ready for single node and misbehaving.

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml
Comment (omertuc) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Right, see https://github.com/openshift/release/pull/14756#discussion_r555354155

**Comment by omertuc** [comment]
> We don't manually generate those test files anymore... Basically you update the master.yaml file I linked to above then run:
> 
> $ make update
> 
> then fix the fields as i suggested..

I didn't manually generate, I put:
```
- as: e2e-aws-single-node
  commands: TEST_SUITE=openshift/conformance/parallel run-tests
  openshift_installer:
    cluster_profile: aws
```

In the config file you linked, generated, then deleted the above snippet.

The reason I deleted it is because:
```
- name: CLUSTER_VARIANT
- value: single-node
```

are manually added to the job, and they get destroyed when you try to re-generate.

If you know how I can specify the CLUSTER_VARIANT template parameter in the config file, it'll be much better, otherwise I can't think of any other solution other than removing the config that was used to generate so the snippet above doesn't get overridden.

**Comment by kikisdeliveryservice** [comment]
thanks @omertuc I'm just going to ask around and get this double checked before we merge. thanks for explaining.

**Comment by omertuc** [comment]
> thanks @omertuc I'm just going to ask around and get this double checked before we merge. thanks for explaining.

Great, keep me posted.

**Comment by kikisdeliveryservice** [comment]
Will look into this later today and if I can't figure out a better solution won't block merging.

In the meantime

/retest

**Comment by kikisdeliveryservice** [comment]
The selection seems to be coming from : 
https://docs.ci.openshift.org/docs/how-tos/contributing-openshift-release/#tolerated-changes-to-generated-jobs



@stevekuznetsov are there any major issues implementing this job as @omertuc has done ? Or we'd need to tolerate that change which seems like a bigger thing. My one concern would be - will this job get picked up for future branches too (unsure how those are generated)?

File: core-services/template-deprecation/_allowlist.yaml
Comment (stevekuznetsov) [inline core-services/template-deprecation/_allowlist.yaml:766]: No. Here, you are adding this new job with an "unknown" blocker. Please either:

 - clearly document the blocker for migrating this type of job to multi-stage steps
 - write a multi-stage job

**Review by stevekuznetsov — CHANGES_REQUESTED** [review]

File: core-services/template-deprecation/_allowlist.yaml
Comment (stevekuznetsov) [reply core-services/template-deprecation/_allowlist.yaml:766]: xref: https://docs.ci.openshift.org/docs/how-tos/migrating-template-jobs-to-multistage/#openshift-installer

**Comment by kikisdeliveryservice** [comment]
just adding a hold until this is updated

/hold

File: core-services/template-deprecation/_allowlist.yaml
Comment (omertuc) [reply core-services/template-deprecation/_allowlist.yaml:766]: - write a multi-stage job

Went with this option eventually. Please review #15187 so I can modify this current PR to use multi-stage.

**Comment by kikisdeliveryservice** [comment]
@stevekuznetsov could you PTAL?

**Comment by omertuc** [comment]
> @stevekuznetsov could you PTAL?
> 
> This seems correct to me, but should pj-reherse pass (I'm not totally sure?)

always run, false optional true. We don't expect it to pass at this point. There are still a lot of fixes needed in the various operators for this to work.

**Comment by kikisdeliveryservice** [comment]
> > @stevekuznetsov could you PTAL?
> > This seems correct to me, but should pj-reherse pass (I'm not totally sure?)
> 
> always run false, optional true. We don't expect it to pass at this point. There are still a lot of fixes needed in the various operators for this to work.

yes i see that :smile:  

but I'd like his review to double check  that these tests are now fine.

**Review by kikisdeliveryservice — APPROVED** [review]
spoke with stevek and this is good to go.

/lgtm

**Comment by kikisdeliveryservice** [comment]
@omertuc looks like there's an error in 

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/14756/pull-ci-openshift-release-master-ci-operator-config-metadata/1354525915652034560

That needs to be resolved?

**Comment by omertuc** [comment]
> @omertuc looks like there's an error in
> 
> [prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/14756/pull-ci-openshift-release-master-ci-operator-config-metadata/1354525915652034560](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/14756/pull-ci-openshift-release-master-ci-operator-config-metadata/1354525915652034560)
> 
> That needs to be resolved?

Right, fixed now, should be okay

**Review by kikisdeliveryservice — APPROVED** [review]
thanks for the update !

/lgtm

**Pull request merged**
